src/Views/GuestSettingsView.vala

```diff
@@ -18,51 +18,34 @@
 */
 
 namespace SwitchboardPlugUserAccounts.Widgets {
-    public class GuestSettingsView : Gtk.Grid {
+    public class GuestSettingsView : Granite.SimpleSettingsPage {
         public signal void guest_switch_changed ();
 
         public GuestSettingsView () {
             Object (
-                column_spacing: 12,
-                margin: 24,
-                row_spacing: 12
+                activatable: true,
+                description: "%s %s".printf (
+                    _("The Guest Session allows someone to use a temporary default account without a password."),
+                    _("Once they log out, all of their settings and data will be deleted.")),
+                icon_name: "avatar-default",
+                title: _("Guest Session")
             );
         }
 
         construct {
-            var image = new Granite.Widgets.Avatar.with_default_icon (64);
-            image.halign = Gtk.Align.END;
-            image.valign = Gtk.Align.START;
-
-            var header_label = new Gtk.Label (_("Guest Session"));
-            header_label.halign = Gtk.Align.START;
-            header_label.get_style_context ().add_class ("h2");
-
-            var guest_switch = new Gtk.Switch ();
-            guest_switch.halign = Gtk.Align.START;
-
             var guest_autologin_switch = new Gtk.Switch ();
             guest_autologin_switch.halign = Gtk.Align.START;
 
             var guest_autologin_label = new Gtk.Label (_("Log In automatically:"));
             guest_autologin_label.xalign = 0;
 
-            var label = new Gtk.Label ("%s %s".printf (
-                _("The Guest Session allows someone to use a temporary default account without a password."),
-                _("Once they log out, all of their settings and data will be deleted.")));
-            label.wrap = true;
-            label.xalign = 0;
-
-            attach (image, 0, 0, 1, 3);
-            attach (header_label, 1, 0, 1, 1);
-            attach (guest_switch, 1, 1, 1, 1);
-            attach (label, 1, 2, 1, 1);
-            attach (guest_autologin_label, 0, 3, 1, 1);
-            attach (guest_autologin_switch, 1, 3, 1, 1);
+            content_area.attach (guest_autologin_label, 0, 3, 1, 1);
```

Comment:
Shouldn't the row number be 0 instead of 3?

Reply:
Oops you're right, I was lazy here